### PR TITLE
Dapp section & visibility changes

### PR DESCRIPTION
**Pull request by jacogr**
- Don't display external/network dapps by default, Closes https://github.com/ethcore/parity/issues/3308
- Split selection into sections based on origin, disclaimer/description for each, Closes https://github.com/ethcore/parity/issues/3307
- Add labels to dapps displaying origin (local, builtin & network)

https://www.youtube.com/watch?v=rCQGnQX8P2Q&feature=youtu.be

### CODE REVIEW

**Comment by gavofyork** [comment]
user should be made aware of external dapps' existence right on the applications page.

**Comment by jacogr** [comment]
Updated with app overlay & section split. https://youtu.be/bv5f9H7qGpc

**Comment by gavofyork** [comment]
> user should be made aware of external dapps' existence right on the applications page.

**Comment by gavofyork** [comment]
interaction looks good.

**Comment by gavofyork** [comment]
js tests look to be failing?

**Comment by jacogr** [comment]
Out of space - 

```
cp: error writing './.dist/Cargo.toml': No space left on device
cp: failed to extend './.dist/Cargo.toml': No space left on device
cp: error writing './.dist/build.rs': No space left on device
cp: failed to extend './.dist/build.rs': No space left on device
cp: error writing './.dist/src/lib.rs': No space left on device
cp: failed to extend './.dist/src/lib.rs': No space left on device
cp: error writing './.dist/src/lib.rs.in': No space left on device
cp: failed to extend './.dist/src/lib.rs.in': No space left on device
/builds/Mirrors/ethcore-parity
ERROR: Build failed (system failure): API error (500): write /var/lib/docker/containers/be79b14b7c1ef0681e1e724c9c6046263cd0d2d5c570b04b4b581a7aaed3c307/hash666675087: no space left on device
```

Will alert @General-Beck

File: js/src/views/Dapps/dapps.css [hunk 1]
```diff
@@ -29,3 +30,25 @@
   flex: 0 1 50%;
   box-sizing: border-box;
 }
+
+.overlay {
+  background: rgba(0, 0, 0, 0.85);
+  bottom: 0.5em;
+  left: -0.25em;
```
Comment (ngotchac) [inline js/src/views/Dapps/dapps.css]: This looks off on my screen (h-scroll bars appears). I think it should be `0.125em` instead

File: js/src/views/Dapps/dappsStore.js [hunk 2]
```diff
@@ -57,14 +83,48 @@ export default class DappsStore {
...
 
   @action showApp = (id) => {
-    this.hiddenApps = this.hiddenApps.filter((_id) => _id !== id);
-    this._writeHiddenApps();
+    this.displayApps = Object.assign({}, this.displayApps, { [id]: { visible: true } });
+    this.writeDisplayApps();
+  }
+
+  @action readDisplayApps = () => {
+    this.displayApps = store.get(LS_KEY_DISPLAY) || {};
```
Comment (ngotchac) [inline js/src/views/Dapps/dappsStore.js]: Shouldn't it be a `boolean` instead of `{}` by default ?

**Comment by ngotchac** [comment]
Looks good. Only remark is that users might not want to display all external dapps by default... ? Or at least have a Select All / Deselect All button(s)

File: js/src/views/Dapps/dappsStore.js (hunk 2 above)
Comment (jacogr) [inline js/src/views/Dapps/dappsStore.js]: It is an object, so in this case when there is none, return the empty object.

**Pull request merged**